models/user.js

```diff
@@ -1,10 +1,148 @@
 const mongoose = require('mongoose');
+const bcrypt = require('bcrypt');
 
-const User = new mongoose.Schema({
-  name: String,
-  hash: String,
-  salt: String,
-  cookie: String,
+const {
+	Schema,
+} = mongoose;
+
+const fields = {
+	firstname: {
+		type: String,
+		required: true,
+	},
+	lastname: {
+		type: String,
+		required: true,
+	},
+	email: {
+		type: String,
+		trim: true,
+		lowercase: true,
+		required: true,
+	},
+	password: {
+		type: String,
+		required: true,
+	},
+};
+
+const user = new Schema(fields, {
+	timestamps: true,
+	toJSON: {
+		virtuals: true,
+	},
+	toObject: {
+		virtuals: true,
+	},
 });
 
-module.exports = mongoose.model('User', User);
+const blacklistFields = ['password'];
```

Comment:
removing fields from the body is not the best way to go about controlling what fields go into the new User object. please refer to comment below on how to structure the new user.